Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAuditdPlugin.py

```diff
@@ -49,94 +68,299 @@ def Get_Marshall(Ensure):
         retd[k] = ld[k]
     return retval, retd
 
-def Set(Ensure):
-    # Add/Remove omsagent config for auoms
-    if Ensure == "Present":
-        text = ReadConf(RESOURCE_OMSAGENT_CONF)
-        WriteConf(OMSAGENT_CONF, text)
-    else:
-        RemoveConf()
+def IsUUID(uuidStr):
+    try:
+        uuidOut = uuid.UUID(uuidStr)
+    except:
+        return False
+    return str(uuidOut).encode('utf-8') == str(uuidStr).encode('utf-8')
+
+def IsValidWorkspaceId(WorkspaceId):
+    if not IsUUID(WorkspaceId):
+        return False
+    return os.path.isdir(os.path.join(AGENT_VAR_ROOT, WorkspaceId))
+
+def GetTmpDir(WorkspaceId):
+    return os.path.join(AGENT_VAR_ROOT, WorkspaceId, "tmp")
+
+def GetAgentConfDir(WorkspaceId):
+    return os.path.join(AGENT_ETC_ROOT, WorkspaceId, "conf/omsagent.d")
+
+def GetOMSAgentConfPath(WorkspaceId):
+    return os.path.join(GetAgentConfDir(WorkspaceId), "auditd_plugin.conf")
+
+def GetTmpAuomsOutputConfPath(WorkspaceId):
+    return os.path.join(GetTmpDir(WorkspaceId), "auoms.conf")
+
+def GetAuomOutputConfPath(WorkspaceId):
+    return os.path.join(AUOMS_OUTCONF_DIR, WorkspaceId + ".conf")
+
+def GetTmpAuditRulesPath(WorkspaceId):
+    return os.path.join(GetTmpDir(WorkspaceId), "auditd_plugin.rules")
+
+def Set(WorkspaceId, Ensure):
+    if not IsValidWorkspaceId(WorkspaceId):
+        return [1]
+
+    (audit_version, plugin_state,
+     actual_audit_rules, actual_outconf, actual_plugin_conf) = GetState(WorkspaceId)
```

Comment:
If you do indent, here, I would suggest at least two spaces. Not sure what Python convention is, though.